File: Csharp/Lab8/Lab8/Lab8/cng/Model/Collections/neww/queue/QueueInterface.cs
using System.Collections.ObjectModel;

namespace Lab8.cng.Model.Collections.neww.queue
{
    public interface IQueueInterface<T> : ICngCollection<T>
    {
        void Enqueue(T t);
        T Dequeue();
        T Peek();
        int Size();
        bool IsEmpty();
        void AddAll(Collection<T> c);

    }
}

File: Kotlin/Lab8/src/by/bntu/fitr/povt/cng/model/collections/neww/queue/QueueInterface.java
package by.bntu.fitr.povt.cng.model.collections.neww.queue;

import by.bntu.fitr.povt.cng.model.collections.neww.CngCollection;

import java.util.Collection;

public interface QueueInterface<T> extends CngCollection<T> {

    void enqueue(T t);
    T dequeue();
    T peek();


    //boolean isFull();

}
